portal/v2/src/ClusterDetailListComponents/Statistics/StatisticsWrapper.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect } from "react"
+import { AxiosResponse } from 'axios';
+import { ICluster } from "../../App"
+import { StatisticsComponent } from './StatisticsComponent';
+import { FetchStatistics } from '../../Request';
+import { apiStatisticsKey, dnsStatisticsKey, ingressStatisticsKey, kcmStatisticsKey } from "../../ClusterDetail";
+import { IMessageBarStyles, MessageBar, MessageBarType, Stack } from '@fluentui/react';
+
+
+export interface IMetricValue {
+  timestamp: Date
+  value: number
+}
+
+export interface IMetrics {
+  Name: string
+  MetricValue: IMetricValue[]
+}
+
+export function StatisticsWrapper(props: {
+  currentCluster: ICluster,
+  detailPanelSelected: string,
+  loaded: boolean,
+  statisticsName: string,
+  duration: string,
+  endDate: Date,
+  graphHeight: number,
+  graphWidth: number,
+}) {
+  const [data, setData] = useState<any>([])
+  const [error, setError] = useState<AxiosResponse | null>(null)
+  const [metrics, setMetrics] = useState<IMetrics[]>([])
+  const [fetching, setFetching] = useState("")
+  const [localDuration, setLocalDuration] = useState(props.duration)
+  const [localEndDate, setLocalEndDate] = useState(props.endDate)
+  const errorBarStyles: Partial<IMessageBarStyles> = { root: { marginBottom: 15 } }
+  const statisticsKeys = [apiStatisticsKey, dnsStatisticsKey, ingressStatisticsKey, kcmStatisticsKey]
+  
+  const errorBar = (): any => {
+    return (
+      <MessageBar
+        messageBarType={MessageBarType.error}
+        isMultiline={false}
+        onDismiss={() => setError(null)}
+        dismissButtonAriaLabel="Close"
+        styles={errorBarStyles}
+      >
+      {error?.statusText}
+      </MessageBar>
+    )
+  }
+
+  // updateData - updates the state of the component
+  // can be used if we want a refresh button.
+  // api/clusterdetail returns a single item.
+  const updateData = (newData: any) => {    
+    setData(newData)
+    const metrics: IMetrics[] = []
+    newData.forEach((element: { metricname: any;
+                                metricvalue: IMetricValue[];}) => {
+      const metric: IMetrics = {
+        Name: element.metricname,
+        MetricValue: element.metricvalue,
+      }
+      metrics.push(metric)
+    });
+    setMetrics(metrics)   
+  }
+
+  useEffect(() => {    
+    const onData = (result: AxiosResponse | null) => {
+      if (result?.status === 200) {        
+        updateData(result.data)
```

Comment:
The error doesn't appear to get cleared when data is returned.

Reply:
Oh yaa, I too noticed that while I was working on your review. setting the error null at receiving status 200.